cpp/src/parquet/column_writer.cc

```diff
@@ -271,7 +271,12 @@ class SerializedPageWriter : public PageWriter {
   }
 
   int64_t WriteDictionaryPage(const DictionaryPage& page) override {
-    int64_t uncompressed_size = page.size();
+    int64_t uncompressed_size = page.buffer()->size();
+    if (uncompressed_size > std::numeric_limits<int32_t>::max()) {
+      throw ParquetException(
+          "Uncompressed dictionary page size overflows to INT32_MAX. Size:",
```

Comment:
Grammar is weird here. Would this be better?
```suggestion
          "Uncompressed dictionary page size overflows INT32_MAX. Size:",
```